package/MDAnalysis/coordinates/H5MD.py

```diff
@@ -0,0 +1,420 @@
+            if 'lambda' in particle_group['data']:
+                ts.data['lambda'] = particle_group['data/lambda/value'][frame]
+            if 'dt' in particle_group['data']:
+                ts.data['dt'] = particle_group['data/dt/value'][frame]
+
+        # set frame box dimensions
+        # set triclinic box vectors
+        ts._unitcell[:] = particle_group['box/edges/value'][frame, :]
+
+        # set particle positions
+        frame_positions = particle_group['position/value'][frame, :]
+        n_atoms_now = frame_positions.shape[0]
+        if n_atoms_now != self.n_atoms:
+            raise ValueError("Frame {} has {} atoms but the initial frame"
+                             " has {} atoms. MDAnalysis is unable to deal"
+                             " with variable topology!"
+                             "".format(frame, n_atoms_now, self.n_atoms))
+        else:
+            ts.positions = frame_positions
+
+        # set particle velocities
+        if self.has_velocities:
+            frame_velocities = particle_group['velocity/value'][frame, :]
+            n_atoms_now = frame_velocities.shape[0]
+            if n_atoms_now != self.n_atoms:
+                raise ValueError("Frame {} has {} atoms but the initial frame"
+                                 " has {} atoms. MDAnalysis is unable to deal"
+                                 " with variable topology!"
+                                 "".format(frame, n_atoms_now, self.n_atoms))
+            else:
+                ts.velocities = frame_velocities
+
+        # set particle forces
+        if self.has_forces:
+            frame_forces = particle_group['force/value'][frame, :]
+            n_atoms_now = frame_forces.shape[0]
+            if n_atoms_now != self.n_atoms:
+                raise ValueError("Frame {} has {} atoms but the initial frame"
+                                 " has {} atoms. MDAnalysis is unable to deal"
+                                 " with variable topology!"
+                                 "".format(frame, n_atoms_now, self.n_atoms))
+            else:
+                ts.forces = frame_forces
+
+        # unit conversion
+        if self.convert_units:
+            ts.time = self.convert_time_from_native(ts.time)
+            self.convert_pos_from_native(ts.dimensions[:3])
+            self.convert_pos_from_native(ts.positions)
+            if self.has_velocities:
+                self.convert_velocities_from_native(ts.velocities)
+            if self.has_forces:
+                self.convert_forces_from_native(ts.forces)
+
+        return ts
+
+    def _read_next_timestep(self):
+        """read next frame in trajectory"""
+        return self._read_frame(self._frame + 1)
+
+    def _translate_h5md_units(self):
+        """converts units from H5MD to MDAnalysis notation
+        and fills units dictionary"""
+
+        if 'units' in self._particle_group['position/time'].attrs:
```

Comment:
Is "units" required by the H5MD standard? If yes, you can fail right here.

If is not required and there's no unit given, then the question is what unit MDAnalysis _should_ assign. What is "normal" in H5MD files? 

Or should we warn (warning and logger.warn()), set it to the MDA default of ps, and also set the `convert_units` kwarg to `False`?

Reply:
I emailed @pdebuyl about whether there was a "standard" unit that H5MD files use, and he confirmed that there's no default unit system and it's entirely up to the person writing the file/running the simulation. I can follow up and see if there's a "normal" unit system that most people using the format conform to

Reply:
The documentation doesn't mention anywhere that units are necessarily _required_, but it wouldn't make sense for a physical simulation to not have units. I think having MDA require units is the safest way. Would incorrectly guessing units be a big deal?

Reply:
Units are not required because for simulation data, there is a lot of dimensionless cases. Of course, there is a consistent set of physical quantities but there is no standard for such systems. We defined the "Units" module so that physical quantities would be defined in the same way in all H5MD files in which such units are employed. MDAnalysis could require the users to specify the unit system upon reading, as an argument to the read function, if the units are not defined in the file.